src/account-db.js

```diff
@@ -38,74 +45,159 @@ export function getLoginMethod(req) {
+   FROM users
+   WHERE users.user_name <> '' and users.owner = 1`,
+      ) || {};
+
+    if (!openIdEnabled || countOfOwner > 0) {
+      if (!needsBootstrap()) {
+        accountDb.mutate('ROLLBACK');
+        return { error: 'already-bootstrapped' };
+      }
+    }
+
+    if (!passEnabled && !openIdEnabled) {
+      accountDb.mutate('ROLLBACK');
+      return { error: 'no-auth-method-selected' };
+    }
+
+    if (passEnabled && openIdEnabled) {
+      accountDb.mutate('ROLLBACK');
+      return { error: 'max-one-method-allowed' };
+    }
+
+    if (passEnabled) {
+      let { error } = bootstrapPassword(loginSettings.password);
+      if (error) {
+        accountDb.mutate('ROLLBACK');
+        return { error };
+      }
+    }
+
+    if (openIdEnabled) {
+      let { error } = await bootstrapOpenId(loginSettings.openId);
+      if (error) {
+        accountDb.mutate('ROLLBACK');
+        return { error };
+      }
+    }
+
+    accountDb.mutate('COMMIT');
+    return passEnabled ? loginWithPassword(loginSettings.password) : {};
+  } catch (error) {
+    accountDb.mutate('ROLLBACK');
+    throw error;
   }
+}
 
-  // Hash the password. There's really not a strong need for this
-  // since this is a self-hosted instance owned by the user.
-  // However, just in case we do it.
-  let hashed = hashPassword(password);
-  accountDb.mutate('INSERT INTO auth (password) VALUES (?)', [hashed]);
-
-  let token = uuid.v4();
-  accountDb.mutate('INSERT INTO sessions (token) VALUES (?)', [token]);
+export function isAdmin(userId) {
+  return hasPermission(userId, 'ADMIN');
+}
 
-  return { token };
+export function hasPermission(userId, permission) {
+  return getUserPermission(userId) === permission;
 }
 
-export function login(password) {
-  if (password === undefined || password === '') {
-    return { error: 'invalid-password' };
+export async function enableOpenID(loginSettings) {
+  if (!loginSettings || !loginSettings.openId) {
+    return { error: 'invalid-login-settings' };
   }
 
-  let accountDb = getAccountDb();
-  let row = accountDb.first('SELECT * FROM auth');
-
-  let confirmed = row && bcrypt.compareSync(password, row.password);
-
-  if (!confirmed) {
-    return { error: 'invalid-password' };
+  let { error } = (await bootstrapOpenId(loginSettings.openId)) || {};
+  if (error) {
+    return { error };
   }
 
-  // Right now, tokens are permanent and there's just one in the
-  // system. In the future this should probably evolve to be a
-  // "session" that times out after a long time or something, and
-  // maybe each device has a different token
-  let sessionRow = accountDb.first('SELECT * FROM sessions');
-  return { token: sessionRow.token };
+  getAccountDb().mutate('DELETE FROM sessions');
 }
 
-export function changePassword(newPassword) {
-  if (newPassword === undefined || newPassword === '') {
-    return { error: 'invalid-password' };
+export async function disableOpenID(
+  loginSettings,
+  checkForOldPassword = false,
```

Comment:
we used to have the possibility to enable openid from the config file. Since we changed the flow to need the user to bootstrap with password once, then enable openid, this was removed, but I forgot to remove this parameter